python/import_util.py

```diff
@@ -122,13 +128,244 @@ def json(ctx: mgp.ProcCtx, path: str) -> mgp.Record():
+        return str(text)
+    if type == "int" or type == "long":
+        return int(text)
+    if type == "boolean":
+        return bool(text)
+    if type == "float" or type == "double":
+        return float(text)
+    if type is None:
+        return text
+
+
+def cast(
+    text: str, type: str, is_list: str
+) -> Union[List[Any], str, int, bool, float]:
+    if is_list is not None:
+        casted_list = list()
+        for element in ast.literal_eval(text):
+            casted_list.append(cast_element(element, is_list))
+        return casted_list
+    return cast_element(text, type)
+
+
+def set_default_keys(key_dict: Dict[str, Any], properties: Dict[str, Any]):
+    for key, value in key_dict.items():
+        if value[3] is not None:
+            properties.update({value[0]: cast(value[3], value[1], value[2])})
+
+
+def set_default_config(config: mgp.Map):
+    if not config.get("readLabels"):
+        config.update({"readLabels": False})
+    if not config.get("defaultRelationshipType"):
+        config.update({"defaultRelationshipType": "RELATED"})
+    if not config.get("storeNodeIds"):
+        config.update({"storeNodeIds": False})
+    if not config.get("source"):
+        config.update({"source": {}})
+    if not config.get("target"):
+        config.update({"target": {}})
+    if (
+        not isinstance(config.get("readLabels"), bool)
+        or not isinstance(config.get("defaultRelationshipType"), str)
+        or not isinstance(config.get("storeNodeIds"), bool)
+        or not isinstance(config.get("source"), dict)
+        or not isinstance(config.get("target"), dict)
+        or (
+            config.get("source") and "label" not in config.get("source").keys()
+        )
+        or (
+            config.get("target") and "label" not in config.get("target").keys()
+        )
+    ):
+        raise TypeError(
+            "Config parameter must be a map with specific \
+             keys and values described in documentation."
+        )
+
+
+@mgp.write_proc
+def graphml(
+    ctx: mgp.ProcCtx,
+    path: str = "",
+    config: mgp.Map = {},
+) -> mgp.Record(status=str):
+    """
+    Procedure to export the whole database to a graphML file.
+
+    Parameters
+    ----------
+    path : str
+        Path to the graphML file containing the exported graph database.
+    config : Map
+
+    """
+
+    set_default_config(config)
+
+    try:
+        tree = ET.parse(path)
+    except Exception:
+        raise OSError("Could not open/read file.")
+
+    root = tree.getroot()
+    graphml_ns = root.tag.split("}")[0].strip("{")
+    namespace = {"graphml": graphml_ns}
+
+    node_keys = dict()
+    rel_keys = dict()
+
+    for key in root.findall(".//graphml:key", namespace):
+        # key value legend: value[0] = attr.name
+        #                   value[1] = attr.type
+        #                   value[2] = attr.list
+        #                   value[3] = default
+        # -> should I make a class out of it?
```

Comment:
I think a `StrEnum` would be handy: https://docs.python.org/3/library/enum.html#enum.StrEnum

Reply:
why strEnum? how would i use that? i can only think of using int enum for clearer access to array indexs

Reply:
Should’ve been clearer, my idea was to make a StrEnum for `"attr.type"` etc. and an IntEnum for the 0...3 indices.